lib/ezutils/classes/ezhttptool.php

```diff
@@ -585,9 +585,17 @@ static function createRedirectUrl( $path, $parameters = array() )
         if ( $port )
             $uri .= ':' . $port;
         $uri .= $path;
-        
-        if ( isset($_SERVER[QUERY_STRING]) && $_SERVER[QUERY_STRING] != '' )
-        	$uri .= '?' . $_SERVER[QUERY_STRING];
+
+        if ( isset($_SERVER['QUERY_STRING']) && $_SERVER['QUERY_STRING'] != '' ) {
+            //If the QUERY_STRING only has "redirect" parameter, do not show it in url
+            if (strpos($_SERVER['QUERY_STRING'],'redirect') === false || strpos($_SERVER['QUERY_STRING'],'&')!== false ) {
```

Comment:
I would have used parse_url() to make sure "redirect" is the only query parameter in the url, looking for "&" is a lot less readable here...